src/meson.build

```diff
@@ -5,6 +5,30 @@ subdir('cps')
 
 cps_include_dir = include_directories('.')
 
+flex = find_program('flex', 'win_flex', version : '>=2.6')
+bison = find_program('bison', 'win_bison', version : '>=2.6')
+
+scanner_gen = generator(flex,
+    output : [ '@BASENAME@.scanner.cpp', '@BASENAME@.scanner.hpp' ],
+    arguments : [
+        '--outfile=@OUTPUT0@',
+        '--header-file=@OUTPUT1@',
+        '@INPUT@'
+    ]
+)
+
+parser_gen = generator(bison,
+    output : [ '@BASENAME@.parser.cpp', '@BASENAME@.parser.hpp' ],
+    arguments : [
+        '-d', '@INPUT@', '-v',
+        '--output=@OUTPUT0@',
+        '--defines=@OUTPUT1@'
+    ]
+)
+
+scanner_files = scanner_gen.process('cps/pc_compat/pc.l')
+parser_files = parser_gen.process('cps/pc_compat/pc.y')
```

Comment:
Generator is a trap, you probably don't want to use that. `generator` really should be called `custom_compiler`, because that's what it's actually for. So the generator writes it's files into the target's private directory (which is also where things like .o files get written).

Meson by design doesn't allow you to put outputs in any directory except the one you're currently in or the target's private directory, so if you want to put these in `cps/pc_compat.` you'll need to put the `custom_target` calls in `cps/pc_compat/meson.build`

Something more like (with the understanding that I've had the misfortune to work with flex and bison on Windows):
```meson
_flex_version = '>= 2.6'
_bison_version = '>= 2.6'
prog_flex = find_program('', required : false)
flex_args = []
prog_binson = find_program('', required : false)
if host_machine.system() == 'windows'
  prog_flex = find_program('win_flex', required : false, version : _flex_vesrion)
  if prog_flex.found()
    # This uses <io.h> instead of <unistd.h>
    flex_args = ['--wincompat']
  endif

  prog_bison = find_program('win_bison', required : false, version : _bison_version)
endif
if not prog_flex.found()
  prog_flex = find_program('flex, 'lex, version : _flex_version)
endif
if not prog_bison.found()
  prog_bison = find_program('bison', version : _bison_version)
endif

pc_scanner = custom_target(
  'pc_scanner,
  command : [prog_flex, flex_arguments, '--outfile=@OUTPUT0@', '--header-file=@OUTPUT1@', '@INPUT@'],
  input : 'pc.l',
  output : ['@BASENAME@.parser.cpp', '@BASENAME@.parser.hpp],
)

pc_parser = custom_target(
  'pc_parser',
  command : [prog_bison, '-d', '@INPUT@', '-v', '--output=@OUTPUT0@', '--defines=@OUTPUT1@'],
  input : 'pc.y',
  output : ['@BASENAME@.parser.cpp', '@BASENAME@.parser.hpp', 'locations.hpp']
)
```

We want to define the `locations.hpp` name even though we're not explicitly writing it, or we can end up with some weird cleanup/rebuild interactions.

When we care more about other OSes, I have a whole pile of "how to make this work with byacc and such. One of these days I'll get around to writing a Meson helper for this...

Reply:
Ahh, I got too excited when I saw the word `generator`. That worked beautifully with use `custom_target`

Reply:
Thank you!